### ISSUE TITLE
Indentation for USING not behaving as expected

### ISSUE THREAD

**Issue by barrywhart** [alanmcruickshank/sqlfluff#199]
Input:
```
SELECT
    user_id
FROM
    audience_counts
JOIN
    `gdpr_safe_users`
USING
    (user_id)
```

Result:
```
$ sqlfluff lint --dialect bigquery sql/audience_size_queries/audience_size_buckets[user_id,list_id].sql
== [sql/audience_size_queries/audience_size_buckets[user_id,list_id].sql] FAIL
L:   7 | P:   1 | L003 | Indentation not consistent with line #6
L:   8 | P:   5 | L003 | Indent expected and not found compared to line #6
```

Is this the intended behavior? If I recall correctly, there was an earlier PR to prefer the indentation seen above.

If I change it to this (i.e. indenting the `USING` clause), no warnings are produced:
```
SELECT
    user_id
FROM
    audience_counts
JOIN
    `gdpr_safe_users`
    USING
        (user_id)
```

**Comment by alanmcruickshank** [comment]
I think this *is* the intended behaviour, and it would be the same as with an `ON` clause. The PR you mentioned was a discussion about whether the `JOIN` keyword should be indented relative to the `FROM` keyword (consensus was no), but I think that `ON` and `USING` should be indented from `JOIN`, because they are something that refers to that join.

That being said, this might be another convention which might be hard to override - do you know what some of the other sql format guidelines say about this one?

**Comment by barrywhart** [comment]
After thinking more about it, I'm growing to like the behavior, although it was initially surprising. I'll see if I can find any opinions on this in either direction, but happy to just close it if you like.

**Comment by barrywhart** [comment]
Same level:
* https://knowledgebase.apexsql.com/rules-of-sql-formatting-nesting-and-indenting-sql-statements/

This Stack Overflow [post](https://stackoverflow.com/questions/272210/sql-statement-indentation-good-practice) has a broad mix of opinions.

I'll close this since it was a misunderstanding and I don't have strong feelings either way. There are other areas of greater importance.
